### PR TITLE
KAT-151: evaluate and prototype desktop shell baseline

### PR DESCRIPTION
**Pull request by gannonh**
## Summary
- evaluate `@kata-shadcn/application-shell1` through `application-shell10` against the current desktop shell and full command-center scope, then record the ranked recommendation in the design doc
- select `@kata-shadcn/application-shell1` as the baseline and integrate a scoped Tauri desktop proof of concept with a shell frame, breadcrumb header, and primary sidebar nav
- add the implementation plan and verification notes for the accepted spike, including fresh desktop test, vitest, lint, typecheck, and coverage evidence

## Test Plan
- [x] `pnpm --filter @kata/desktop test`
- [x] `pnpm exec vitest run tests/unit/desktop/layout.test.tsx tests/unit/desktop/sidebar.test.tsx tests/unit/desktop/navigation.test.tsx tests/unit/desktop/store.test.ts`
- [x] `pnpm --filter @kata/desktop lint`
- [x] `pnpm --filter @kata/desktop typecheck`
- [x] `pnpm coverage`
- [x] bounded startup check via `pnpm desktop:tauri:dev`

## Release Notes

* **New Features**
  * Introduced dynamic breadcrumb navigation that displays the currently active application route in the interface, helping users understand their location within the app.

* **Improvements**
  * Enhanced navigation accessibility by implementing proper semantic labels and attributes to better support screen readers and assistive technologies.

<h3>Greptile Summary</h3>

This PR implements a desktop shell baseline after a thorough evaluation of 10 `@kata-shadcn` application shell candidates. The evaluation selected `application-shell1` as the best fit for both current needs and future command-center expansion. The implementation adds a breadcrumb header to the existing sidebar layout while maintaining route-driven rendering through `Outlet`. All changes follow the registry-first selection policy documented in `AGENTS.md`, with comprehensive design documentation, TDD approach, and full verification coverage (tests, lint, typecheck, manual desktop run).

<h3>Confidence Score: 5/5</h3>

- This PR is safe to merge with no identified risks
- Score reflects excellent engineering practices: comprehensive design evaluation with clear rationale, proper adherence to registry-first selection policy, focused scope (shell-level changes only), strong test coverage for new functionality, correct breadcrumb route matching logic with proper `end` default handling, accessibility improvements, and thorough verification steps documented and executed
- No files require special attention

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| apps/desktop/src/components/Layout.tsx | Added breadcrumb header with dynamic route matching using `matchPath`. Implementation correctly defaults `end` to `true` to prevent parent routes from matching child paths. |
| apps/desktop/src/components/Sidebar.tsx | Added accessibility label `aria-label="Primary"` to navigation element. Clean, focused change. |
| docs/plans/2026-02-28-kat-151-desktop-shell-spike-design.md | Comprehensive design evaluation of 10 application shell candidates with clear comparison matrix, ranked recommendation, and rationale. Follows required registry-first selection policy. |
| docs/plans/2026-02-28-kat-151-desktop-shell-spike-implementation.md | Detailed implementation plan with TDD approach, clear task breakdown, and verification steps. Properly scopes changes to shell-level modifications only. |
| tests/unit/desktop/layout.test.tsx | Added tests for shell frame structure and breadcrumb edge cases (unmatched routes). Comprehensive coverage of new functionality. |

</details>

</details>

<sub>Last reviewed commit: 7c3a051</sub>

**Context used:**

- Rule from `dashboard` - CLAUDE.md ([source](https://app.greptile.com/review/custom-context?memory=7030b86d-bc4b-4fd4-b770-9236b06fdf78))
- Rule from `dashboard` - AGENTS.md ([source](https://app.greptile.com/review/custom-context?memory=f27cdf3b-b406-4db4-9c22-e4b12003b532))

### CODE REVIEW

**Comment by gannonh** [comment]
@codex review

File: docs/plans/2026-02-28-kat-151-desktop-shell-spike-design.md [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+- The screenshots in `docs/screenshots/app-shells` are part of the design source of truth for this comparison.
+- `docs/PROJECT_PLAN.md` is a required assessment input because it defines the intended desktop command center and future information density.
+
+## Goals
+
+- Compare all ten application shell candidates using a consistent rubric.
+- Use the no-preview registry workflow with temp installs for direct source inspection.
+- Produce a ranked recommendation with explicit adopt/adapt/reject guidance.
+- Select one shell baseline that fits both the current desktop app and the broader command-center roadmap.
+- Land a same-ticket proof of concept shell swap for the chosen baseline.
```
[the remark this answers is not in the document]
Comment (gannonh) [reply docs/plans/2026-02-28-kat-151-desktop-shell-spike-design.md]: Updated. Replaced all remaining "proof of concept" compound modifiers in the design doc with "proof-of-concept" for consistency.

File: tests/unit/desktop/store.test.ts [hunk 2]
```diff
@@ -7,6 +7,11 @@ describe('useAppStore', () => {
     useAppStore.setState({ sidebarCollapsed: false });
   });
 
+  test('keeps sidebar collapse state in the shared app store', () => {
+    const state = useAppStore.getState();
+    expect(typeof state.sidebarCollapsed).toBe('boolean');
+  });
```
[the remark this answers is not in the document]
Comment (gannonh) [reply tests/unit/desktop/store.test.ts]: Updated. Removed the redundant runtime type assertion test (`keeps sidebar collapse state in the shared app store`) and kept the concrete default-state behavior test.

File: docs/plans/2026-02-28-kat-151-desktop-shell-spike-implementation.md [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+  - Map existing `routes` metadata into the shell nav API
+  - Add one small shell-state field if the chosen shell needs it
+- Rejected candidates:
+  - `application-shell9`: too IDE-specific for the command-center IA
+```
+
+**Step 4: Remove rejected temp installs only after the winner is chosen**
+
+Run: `rm -rf tmp/registry-install/application-shell{1,2,3,4,5,6,7,8,9,10}`
+Expected: temp inspection artifacts are removed; any code copied intentionally for the POC should be reintroduced in production paths separately
```
[the remark this answers is not in the document]
Comment (gannonh) [reply docs/plans/2026-02-28-kat-151-desktop-shell-spike-implementation.md]: Updated. Replaced the blanket delete command with a loop that removes rejected shell temp installs while preserving the selected winner (`X`) for implementation reference.
